src/kOS.Safe/Encapsulation/ScalarValue.cs

```diff
@@ -0,0 +1,456 @@
+﻿using System;
+using kOS.Safe.Encapsulation.Suffixes;
+using kOS.Safe.Exceptions;
+
+namespace kOS.Safe.Encapsulation
+{
+    abstract public class ScalarValue : Structure, IConvertible
+    {
+        protected object internalValue;
```

Comment:
I suppose I can modify the `Value` property to make it have a `protected` setter instead of exposing `internalValue` to subclasses.  In the case of `abstact` classes however it makes sense to me that `protected` fields be allowed, since by definition the `abstact` class itself cannot be instantiated.  I will make the change to `Value`.